### PR TITLE
use hierarchical_memory_limit instead of limit_in_bytes with cgroup v1

### PR DESCRIPTION
**Pull request by Nicholaswang**
Signed-off-by: Nicholaswang <wzhever@gmail.com>

Fixing a bug:
When using limit_percentage config for memory_limiter, it can get an incorrect limit value since the memory.limit_in_bytes is set to unlimited, and it gets the memory limit of the container node:
image

the container node meminfo:
企业微信截图_16467551589129

configure limit_percentage to 75 and spike_limit_percentage to 20:
企业微信截图_16467539509973

with the latest code, it will get the memory limit of the node which is 192658Mb of memory limit:
企业微信截图_1646753927335

after fixing: use hierarchical_memory_limit instead of limit_in_bytes with cgroup v1, it will get the container memory limit: 4096Mb:
企业微信截图_16467541014594

this is a common issue for projects using cgroups memory limit, such as https://github.com/VictoriaMetrics/VictoriaMetrics/issues/84 
VictoriaMetrics also fix this issue with `hierarychical_memory_limit` in `memory.stat`

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/4972

### CODE REVIEW

**Review by dashpole — COMMENTED** [review]
looks good.  Can you extend the unit tests to cover this case?

**Comment by Nicholaswang** [comment]
Sure

**Review by dashpole — APPROVED** [review]

**Comment by Nicholaswang** [comment]
@dashpole unit tests have been extented.

**Comment by Nicholaswang** [comment]
Hi, @bogdandrutu @dashpole 
The pr is still open, do I need to increase unittest coverage?

**Comment by dashpole** [comment]
Can you rebase?  It looks like some tests are failing, and those may have been fixed.

**Comment by jpkrohling** [comment]
PR rebased.

**Comment by Nicholaswang** [comment]
Hi, @dashpole @jpkrohling Do I need to fix the failing tasks before merged?

**Comment by jpkrohling** [comment]
Yes. If they are caused by intermittent tests, we can restart, but it looks like you need to improve the test coverage (if possible) and add a changelog entry. If you don't think a changelog entry is required for this change and the reviewer agrees with you, I can add a label to skip this check.

**Comment by Nicholaswang** [comment]
ok, I will improve the coverage

**Comment by Nicholaswang** [comment]
@jpkrohling added two more cases to imprive coverage.

**Comment by Nicholaswang** [comment]
@dashpole @jpkrohling Hi, seems like it needs to be retriggered

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +29,6 @@
 - Do not set MeterProvider to global otel (#5146)
 - Make `InstrumentationLibrary<signal>ToScope` helper functions unexported (#5164)
 - Remove Log's "ShortName" from logging exporter output (#5172)
-- `exporter/otlp`: Retry RESOURCE_EXHAUSTED only if the server returns RetryInfo (#5147)
```
Comment (dmitryax) [inline CHANGELOG.md:32]: Looks like this PR includes some unrelated changes. Can you please remove them? Also please add a changelog entry

Comment (Nicholaswang) [reply CHANGELOG.md]: sure, caused by wrong revert click after rebase

**Comment by jpkrohling** [comment]
@dashpole, @dmitryax, do you know what's pending for this PR?

**Comment by dashpole** [comment]
It looks like it somehow includes a bunch of unrelated commits.

**Comment by jpkrohling** [comment]
@Nicholaswang, are you still interested in working on this one?
